core/tests/protractor/learnerDashboard.js

```diff
@@ -99,88 +97,90 @@ describe('Learner dashboard functionality', function() {
-    // Create first exploration named 'About Oppia'
+    // Create exploration 'About Oppia'
     createAboutOppiaExploration();
     // Create a second exploration named 'Dummy Exploration'.
-    workflow.createAndPublishExploration(
-      'Dummy Exploration',
-      'Algebra',
-      'To learn about Algebra!',
-      'English'
-    );
     users.logout();
+
     users.login('learner@learnerDashboard.com');
-    // TODO(hoangviet1993): Leave the exploration in between.
-    // The exploration should be found in the 'In Progress' section.
-    // Expecting Alert window after leaving and unable to handle the alert at
-    // the point of writing this.
-    // Issue actively discussed here:
-    // https://github.com/angular/protractor/issues/308
-    // Play an exploration and leave it in between.
-
-    // Play exploration completely.
+    // Play exploration 'About Oppia'.
     libraryPage.get();
     libraryPage.findExploration('About Oppia');
     libraryPage.playExploration('About Oppia');
+    explorationPlayerPage.expectExplorationNameToBe('About Oppia');
     explorationPlayerPage.submitAnswer('Continue', null);
-    explorationPlayerPage.submitAnswer(
-      'MultipleChoiceInput', 'Those were all the questions I had!');
+    explorationPlayerPage.expectExplorationToNotBeOver();
+
+    // Disable waiting for Angular to accept alert.
+    browser.waitForAngularEnabled(false);
+    // Refresh page to simulate user leaving.
+    browser.navigate().refresh().then(function() {
+      browser.wait(until.alertIsPresent(), 5000);
+      return browser.switchTo().alert().then(function (alert) {
+        alert.accept().then(function() {
+          // Re-enable waiting for Angular.
+          return browser.waitForAngularEnabled(true);
+        });
+      });
+    });
 
+    general.waitForLoadingMessage();
+    // Learner Dashboard should display 'About Oppia' as incomplete.
+    learnerDashboardPage.get();
+    learnerDashboardPage.navigateToInCompleteSection();
+    learnerDashboardPage.navigateToIncompleteExplorationsSection();
+    learnerDashboardPage.expectTitleOfExplorationSummaryTileToMatch(
+      'About Oppia');
+
+    // Now play exploration 'About Oppia' completely.
     libraryPage.get();
-    libraryPage.findExploration('Dummy Exploration');
-    libraryPage.playExploration('Dummy Exploration');
-    explorationPlayerPage.expectExplorationNameToBe('Dummy Exploration');
-    explorationPlayerPage.rateExploration(5);
-    general.waitForSystem();
+    libraryPage.findExploration('About Oppia');
+    libraryPage.playExploration('About Oppia');
+    general.waitForLoadingMessage();
+    explorationPlayerPage.expectExplorationNameToBe('About Oppia');
+    explorationPlayerPage.submitAnswer('Continue', null);
+    explorationPlayerPage.submitAnswer(
+      'MultipleChoiceInput', 'Those were all the questions I had!');
 
-    // Both should be added to
-    // the completed section.
+    // Both should be added to the completed section.
```

Comment:
There's only one exploration now, right?

That said ... I don't understand why we removed one? Wouldn't that be a regression in test coverage?

Reply:
Oops! I temporary deleted creating the additional exploration to speed up running the suite and forgot to add it back! Fixed!